addons/mrp_subcontracting_purchase/tests/test_mrp_subcontracting_purchase.py

```diff
@@ -1030,3 +1030,69 @@ def test_mrp_subcontract_modify_date(self):
         with Form(mo) as production_form:
             production_form.date_start = original_mo_start_date
         self.assertEqual(mo.date_start, original_mo_start_date)
+
+    def test_changing_qty_producing_does_not_create_extra_move_lines(self):
+        """This test ensures that during component recording in a subcontracted MO, changing
+        qty_producing takes account of existing reserved move lines and their resupplied lot/quantity
+        to fulfill component consumption, without creating extra lines with empty lots.
+        """
+
+        resupply_sub_on_order_route = self.env['stock.route'].search([('name', '=', 'Resupply Subcontractor on Order')])
```

Comment:
```suggestion
        resupply_sub_on_order_route = self.warehouse.subcontracting_pull_id.route_id
```

`Avoid calling ORM` when it’s not necessary.